### ISSUE TITLE
Geth returning no data for random uncle blocks;  method `eth_getBlockByHash`

### ISSUE THREAD

**Issue by rx294** [ethereum/go-ethereum#25557]
#### System information

Geth version: 1.10.21-stable and 1.10.19-stable
OS & Version: Ubuntu 20.04

#### Expected behaviour

`eth_getBlockByHash` returns all uncle blocks correctly by hash

#### Actual behaviour

Geth nodes randomly misses uncle blocks by method `eth_getBlockByHash`

Behavior seems random; different blocks are missed on different nodes.
Issue happening on full, archive nodes

Nethermind client seems to be only one able to handle all uncle hashes I have tried.

#### Steps to reproduce the behaviour

The following is a full node that has completed syncing

```
curl -s -X POST -H "Content-Type: application/json" --data '{"method":"eth_getBlockByHash","params":["0x43dcef55b1eb2ceb6dd975190812dcc5c1a77598ac4c9325c6ad9b29cff065bf",false],"id":1,"jsonrpc":"2.0"}' http://150.136.140.196:8545

{"jsonrpc":"2.0","id":1,"result":null}
```

**Comment by karalabe** [comment]
Uncle blocks are "side chains". Some may never propagate through the network is a miner includes its own late uncle, but they can still add it as an uncle header. Once the chain progresses enough forward, even propagated side chains get deleted.

TL;DR: Don't rely on the availability of uncle blocks, they will not be available long term and many might not ever be available even in the short term. The header is all you'll have available guaranteed.

**Comment by rx294** [comment]
@karalabe 
Uncle blocks are rewarded ETH. How do we validate that the uncle blocks existed and were correct and are being rewarded to the miner appropriately if we cannot validate the uncle block?

Also, our Nethermind nodes return every single uncle block we've tested.
